src/Controller/Component/PaginatorComponent.php

```
@@ -65,305 +44,60 @@ public function implementedEvents()
-            $field = key($options['order']);
-            $inWhitelist = in_array($field, $options['sortWhitelist'], true);
-            if (!$inWhitelist) {
-                $options['order'] = [];
-            }
-            return $options;
-        }
-
-        $tableAlias = $object->alias();
-        $order = [];
-
-        foreach ($options['order'] as $key => $value) {
-            $field = $key;
-            $alias = $tableAlias;
-            if (is_numeric($key)) {
-                $order[] = $value;
-            } else {
-                if (strpos($key, '.') !== false) {
-                    list($alias, $field) = explode('.', $key);
-                }
-                $correctAlias = ($tableAlias === $alias);
-
-                if ($correctAlias && $object->hasField($field)) {
-                    $order[$tableAlias . '.' . $field] = $value;
-                }
-            }
-        }
-        $options['order'] = $order;
-
-        return $options;
+        return $this->_paginator->config($key, $value, $merge);
     }
 
     /**
-     * Check the limit parameter and ensure it's within the maxLimit bounds.
+     * Using the magic call as a proxy to the refactored paginator.
      *
-     * @param array $options An array of options with a limit key to be checked.
-     * @return array An array of options for pagination
+     * @param string $method Method name.
+     * @param array $args The arguments passed to the called method.
+     * @return mixed
      */
-    public function checkLimit(array $options)
+    public function __call($method, $args)
     {
-        $options['limit'] = (int)$options['limit'];
-        if (empty($options['limit']) || $options['limit'] < 1) {
-            $options['limit'] = 1;
+        if (method_exists($this->_paginator, $method)) {
+            return call_user_func_array([$this->_paginator, $method], $args);
         }
-        $options['limit'] = min($options['limit'], $options['maxLimit']);
```

I was trying to comment on the __call method that silently fails when methods are missing.